packages/logging-interceptor/README.md

```diff
@@ -159,6 +160,92 @@ Error: Internal Server Error
 
 ```
 
+### Masking
+By default, the whole body of the request and of the response is logged. However, the payload may contain sensitive data you would like to hide in the logs. The `Log` decorator provides masking options which allow to override the default properties logged for each endpoint. You just need to provide the path of the properties to mask for the request and/or the response. The corresponding values will be replaced by the placeholder `****` in the resulting log.
+
+For example:
+
+```typescript
+import { Log } from '@algoan/nestjs-logging-interceptor'
+import { Body, Controller } from '@nestjs/common';
+
+@Controller('users')
+export class UsersController {
+
+  @Post()
+  @Log({
+    mask: {
+      request: [
+        'lastName', // simple property
+        'maritalSituation', // object property
+        'contact.phone', // nested property
+        'friends.lastName' // property of an array element
+        ],
+      response: ['lastName'],
+    },
+  })
+  public createUser(@Body() payload: UserDto) {
+    // create and return the new user
+  }
+}
+```
+
+If the endpoint is called with:
+```json
+{
+  "firstName": "John",
+  "lastName": "Doe",
+  "maritalSituation": {
+    "status": "MARRIED",
+    "wife": { "firstName": "Jane", "lastName": "Doe" }
+  },
+  "contact": { "email": "john.doe@email.com", "phone": "+330102030405" },
+  "friends": [
+    { "firstName": "James", "lastName": "Smith" },
+    { "firstName": "Emma", "lastName": "Cole" }
+  ]
+}
+```
+The body in the logged request will be:
+```json
+{
+  "firstName": "John",
+  "lastName": "****",
+  "maritalSituation": "****",
+  "contact": { "email": "john.doe@email.com", "phone": "****" },
+  "friends": [
+    { "firstName": "James", "lastName": "****" },
+    { "firstName": "Emma", "lastName": "****" }
+  ]
+}
+```
+
+If you want you to mask the whole body of the request/response, you can pass `true` instead of the array containing the exhaustive list of properties.
+
+In addition, other options can be set at the interceptor level:
+
+```typescript
+import { Module } from '@nestjs/common';
+import { APP_INTERCEPTOR } from '@nestjs/core';
+import { LoggingInterceptor } from '@algoan/nestjs-logging-interceptor';
+
+@Module({
+  providers: [
+    {
+      provide: APP_INTERCEPTOR,
+      useClass: () => {
+        const interceptor: LoggingInterceptor = new LoggingInterceptor();
+        interceptor.setDisabledMasking(true); // Ignore masking options in the entire applications
+        interceptor.setMaskingPlaceholder("hidden"); // Replace the default placeholder '****' by a custom one
+
+        return interceptor;
+      },
```

Comment:
Is providing the option via interceptor constructor more convient?
Just `return new Interceptor({ disableMasking: true, maskingPlacehokder: "hidden" })`

Reply:
Initially, I wanted to do so but I wanted to be consistent with the `userPrefix` option which cannot be passed in the constructor. That being said, I agree with it is more convenient to pass options as constructor parameters. So I will allow for the three parameters.